jenkinsci/jenkins
# [JENKINS-74858]  Added validation for Password length in FIPS mode

See [JENKINS-74858](https://issues.jenkins.io/browse/JENKINS-74858).

Made changes to ensure that an exisiting user is now unable to alter the password to less than 14 characters in length when operating in FIPS mode.



### Testing done

**Automated Unit Testing**
Implemented unit tests to verify FIPS compliance for password changes:

1. Ensured that passwords shorter than 14 characters are rejected when FIPS mode is enabled.
2. Verified that mismatched passwords between user.password and user.password2 trigger appropriate errors.

**Manual Testing**
Did Manual testing to verify my changes to ensure that everything is working properly.

------------------------------------------------------------
**Without FIPS mode:** Shorter password is being accepted.
![Screenshot 2024-11-20 at 12 37 33 PM](https://github.com/user-attachments/assets/092ecfc3-42a6-4ab5-a7d8-132956eaed6d)


-------------------------------------------------------------
**With FIPS mode:** Shorter password is not being accepted.
![Screenshot 2024-11-20 at 12 41 00 PM](https://github.com/user-attachments/assets/7c48a730-1a7f-47af-b315-1fd5cf279924)
![Screenshot 2024-11-20 at 12 41 09 PM](https://github.com/user-attachments/assets/6deb9c6a-0fc6-4776-9555-0bdabce40c91)




### Proposed changelog entries

- Added Password validation to ensure that existing users cannot create a password of less than 14 characters in length(when in FIPS mode).



### Proposed upgrade guidelines

N/A



```[tasklist]
### Submitter checklist
```

### Desired reviewers

@mention



Before the changes are marked as `ready-for-merge`:

```[tasklist]
### Maintainer checklist
```
